### PR TITLE
fix: Amplitude.build() is not main thread safe on Android

### PR DESCRIPTION
**Pull request by bohan-amplitude**
### Summary

put IO in coroutines for Amplitude#build() on Android

### Checklist

* [x] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-Kotlin/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?no

### CODE REVIEW

**Comment by bohan-amplitude** [comment]
@qingzhuozhen I moved the disk IO into the coroutines for Amplitude#build() based on the customer issue: If called Amplitude#build() on main thread on Android. The Android strict mode will detect IO on main thread and throw DiskReadViolation. Want to check if this will case issue since Amplitude#build() is possible to return before idContainer init

**Comment by bohan-amplitude** [comment]
@liuyang1520 Could you help check the reset id test

**Comment by liuyang1520** [comment]
> @liuyang1520 Could you help check the reset id test

Forked this branch locally, moved the `idContainer` init outside `amplitudeScope`, test is passing for me. Trying to understand the case here, is it possible to init `idContainer` outside `amplitudeScope`? 

```kotlin
        val storageDirectory = (configuration as Configuration).context.getDir("${FileStorage.STORAGE_PREFIX}-${configuration.instanceName}", Context.MODE_PRIVATE)
        idContainer = IdentityContainer.getInstance(
            IdentityConfiguration(
                instanceName = configuration.instanceName,
                apiKey = configuration.apiKey,
                identityStorageProvider = FileIdentityStorageProvider(),
                storageDirectory = storageDirectory
            )
        )
        amplitudeScope.launch(amplitudeDispatcher) {
...
```

File: android/src/main/java/com/amplitude/android/Amplitude.kt [hunk 1]
```diff
@@ -27,13 +27,11 @@ open class Amplitude(
     private lateinit var androidContextPlugin: AndroidContextPlugin
 
     override fun build() {
-        val storageDirectory = (configuration as Configuration).context.getDir("${FileStorage.STORAGE_PREFIX}-${configuration.instanceName}", Context.MODE_PRIVATE)
         idContainer = IdentityContainer.getInstance(
             IdentityConfiguration(
                 instanceName = configuration.instanceName,
                 apiKey = configuration.apiKey,
-                identityStorageProvider = FileIdentityStorageProvider(),
-                storageDirectory = storageDirectory
```
Comment (qingzhuozhen) [inline android/src/main/java/com/amplitude/android/Amplitude.kt:36]: would the identity storage still create correctly in this case? The identity module will trigger the initialization right?

**Review by liuyang1520 — APPROVED** [review]
LGTM

File: android/src/main/java/com/amplitude/android/Amplitude.kt (hunk 1 above)
Comment (bohan-amplitude) [reply android/src/main/java/com/amplitude/android/Amplitude.kt:36]: moved the storage init after set the directory path

**Pull request merged**